Add Async Batch Shard Settings (#7056)
opensearch-project/documentation-website: _install-and-configure/configuring-opensearch/cluster-settings.md

    - `REPLICA_FIRST` – Replica shards are relocated first, before primary shards. This prioritization may help prevent a cluster's health status from going red when carrying out shard relocation in a mixed-version, segment-replication-enabled OpenSearch cluster. In this situation, primary shards relocated to OpenSearch nodes of a newer version could try to copy segment files to replica shards on an older version of OpenSearch, which would result in shard failure. Relocating replica shards first may help to avoid this in multi-version clusters. 
    - `NO_PREFERENCE` – The default behavior in which the order of shard relocation has no importance. 

> - `cluster.allocator.gateway.batch_size` (Integer): Limits the number of shards sent in one batch to data nodes for fetching metadata. Default is `2000`.

Reviewer: can we change it to -> Limits the number of shards sent in one batch to data nodes for **fetching unassigned shard** metadata

> - `cluster.allocator.gateway.batch_size` (Integer): Limits the number of shards we send in one batch to data nodes for fetching metadata. Default is `2000`.

Reviewer 2: ```suggestion
- `cluster.allocator.gateway.batch_size` (Integer): Limits the number of shards sent in one batch to data nodes for fetching metadata. Default is `2000`.
```

> - `cluster.allocator.gateway.batch_size` (Integer): Limits the number of shards sent in one batch to data nodes for fetching unassigned shard metadata. Default is `2000`.

Reviewer 2: ```suggestion
- `cluster.allocator.gateway.batch_size` (Integer): Limits the number of shards sent to data nodes in one batch meant for fetching unassigned shard metadata. Default is `2000`.
```

> - `cluster.allocator.gateway.batch_size` (Integer): Limits the number of shards sent to data nodes in one batch meant for fetching unassigned shard metadata. Default is `2000`.

Reviewer 2: ```suggestion
- `cluster.allocator.gateway.batch_size` (Integer): Limits the number of shards sent to data nodes in one batch for fetching any unassigned shard metadata. Default is `2000`.
```

> - `cluster.allocator.existing_shards_allocator.batch_enabled` (Boolean): Enables batch allocation of unassigned shards already existing on the disk. This allows for sending all `Unassigned`shards to the `ExistingShard` allocator. The allocator decides how to allocate the unassigned shards. Default is `False`.

Reviewer: Enables batch allocation of unassigned shards already existing on the disk as opposed to one shard at a time. This reduces the memory and transport overhead by fetching unassigned shard metadata in a batch call. Default is `False`.

Author: Updated the wordings.

> - `cluster.allocator.existing_shards_allocator.batch_enabled` (Boolean): Boolean setting to enable/disable batch allocation of unassigned shards already existing on disk. This will allow sending all Unassigned Shards to the ExistingShard Allocator to make decision to allocate in one or more go. Default is `False`.

Reviewer 2: ```suggestion
- `cluster.allocator.existing_shards_allocator.batch_enabled` (Boolean): Enables batch allocation of unassigned shards already existing on the disk. This allows for sending all `Unassigned`shards to the `ExistingShard` allocator. The allocator decides how to allocate the unassigned shards. Default is `False`.
```

Author: Addressed changes in next commit. Thanks!

> - `cluster.allocator.existing_shards_allocator.batch_enabled` (Boolean): Enables batch allocation of unassigned shards already existing on the disk as opposed to one shard at a time. This reduces the memory and transport overhead by fetching unassigned shard metadata in a batch call. Default is False.

Reviewer 2: ```suggestion
- `cluster.allocator.existing_shards_allocator.batch_enabled` (Boolean): Enables batch allocation of unassigned shards which already exist on the disk as opposed to one shard at a time. This reduces the memory and transport overhead by fetching any unassigned shard metadata in a batch call. Default is `False`.
```

> - `cluster.allocator.existing_shards_allocator.batch_enabled` (Boolean): Enables batch allocation of unassigned shards which already exist on the disk as opposed to one shard at a time. This reduces the memory and transport overhead by fetching any unassigned shard metadata in a batch call. Default is `False`.

Reviewer 3: ```suggestion
- `cluster.allocator.existing_shards_allocator.batch_enabled` (Boolean): Enables batch allocation of unassigned shards that already exist on the disk as opposed to allocating one shard at a time. This reduces memory and transport overhead by fetching any unassigned shard metadata in a batch call. Default is `False`.
```

> - `cluster.allocator.existing_shards_allocator.batch_enabled` (Boolean): Enables batch allocation of unassigned shards which already exist on the disk as opposed to one shard at a time. This reduces the memory and transport overhead by fetching any unassigned shard metadata in a batch call. Default is `False`.

Reviewer 3: Is `false` intentionally capitalized?

Reviewer 2: Should be lower-cased?
